components/builder-api/src/depot/server.rs

```diff
@@ -1702,27 +1709,50 @@ fn show_package(req: &mut Request) -> IronResult<Response> {
     let mut ident = ident_from_req(req);
     let qualified = ident.fully_qualified();
     let target = target_from_req(req);
+    let config = req.get::<persistent::Read<Config>>().unwrap();
 
     if let Some(channel) = channel {
         if !qualified {
-            let mut request = OriginChannelPackageLatestGet::new();
-            request.set_name(channel.clone());
-            request.set_target(target.to_string());
-            request.set_visibilities(visibility_for_optional_session(
-                req,
-                session_id,
-                &ident.get_origin(),
-            ));
-            request.set_ident(ident.clone());
-
-            match route_message::<OriginChannelPackageLatestGet, OriginPackageIdent>(req, &request)
+            match MemCache::new(config.clone()).get_package(ident.clone().into(), channel.as_str())
```

Comment:
if you pull this up to the top of the function.. 
```
if let Some(id) = MemCache::new(..).get_package(...) {
  return the Ident
}
```
 that should cause fewer changes in the rest of the function below (other than a couple of lines to write out to the cache)